plugins/flytekit-spark/flytekitplugins/spark/task.py

```diff
@@ -25,12 +26,20 @@ class Spark(object):
         hadoop_conf: Dictionary of hadoop conf. The variables should match a typical hadoop configuration for spark
         executor_path: Python binary executable to use for PySpark in driver and executor.
         applications_path: MainFile is the path to a bundled JAR, Python, or R file of the application to execute.
+        driver_pod_template: PodTemplate to use for the driver pod.
+        driver_pod_template_name: Name of the existing PodTemplate resource to use for the driver pod.
+        executor_pod_template: PodTemplate to use for the executor pods.
+        executor_pod_template_name: Name of the existing PodTemplate resource to use for the executor pods.
     """
 
     spark_conf: Optional[Dict[str, str]] = None
     hadoop_conf: Optional[Dict[str, str]] = None
     executor_path: Optional[str] = None
     applications_path: Optional[str] = None
+    driver_pod_template: Optional["PodTemplate"] = None
```

Comment:
This is okay to be `PodTemplate` though.  And we can translate from `PodTemplate` to the model `K8sPod` using this [same logic](https://github.com/flyteorg/flytekit/blob/0e82799e3a701075012a6da76f4a4f5846d5c969/flytekit/core/python_auto_container.py#L205).  This translation would happen in the `get_custom` function below.